packages/mui-system/src/Unstable_Grid/createGrid.tsx

```diff
@@ -162,7 +162,12 @@ export default function createGrid(
         {...other}
       >
         {React.Children.map(children, (child) => {
-          if (React.isValidElement(child) && isMuiElement(child, ['Grid'])) {
+          if (
+            React.isValidElement(child) &&
+            (isMuiElement(child, ['Grid']) ||
+              // eslint-disable-next-line no-underscore-dangle
+              (child.type as any)?._payload?.value?.muiName === 'Grid')
```

Comment:
The fix works for me in a sandbox: https://codesandbox.io/p/sandbox/https-github-com-mui-material-ui-issues-37932-forked-qtf5j2?file=%2Fapp%2Fclient%2Fpage.tsx%3A9%2C1 (I gave the CI a kick)

Maybe we could add this additional check directly inside `isMuiElement`? It does increase the scope of the PR but so far [only 8](https://github.com/search?q=repo%3Amui%2Fmaterial-ui%20unstable_isMuiElement&type=code) Joy UI components (in addition to `Grid`) uses this

Without it other issues occur in a Next.js server component page when it's used, e.g. [this](https://github.com/mui/material-ui/blob/master/packages/mui-joy/src/Typography/Typography.tsx#L241) usage of Joy UI Typography + Skeleton – In the `/server` page, the Skeleton doesn't show without manually adding the `variant` prop: https://codesandbox.io/p/sandbox/https-github-com-mui-material-ui-pull-38129-forked-93y266?file=%2Fapp%2Fserver%2Fpage.tsx%3A15%2C1 CC @siriwatknp

Reply:
Where does the `_payload` object come from? Is it related to generic server rendered component in general or is it specific to nextjs?

Reply:
Probably related to [this](https://github.com/facebook/react/blob/2807d781a08db8e9873687fccc25c0f12b4fb3d4/packages/react/src/ReactLazy.js#L45).

Reply:
@sai6855 I think it's better to fix the `isMuiElement` util in this case. As a note, the `muiName` should be replaced with CSS (likely `:has` selector) but the browser support is not wide enough.

Reply:
Also, please add a comment about `child.type as any)?._payload?.value?.muiName === 'Grid'` with the link @brijeshb42 pointed to.